app/src/main/java/com/douglasbruce/tasky/core/data/repository/EventRepositoryImpl.kt

```diff
@@ -107,4 +113,12 @@ class EventRepositoryImpl @Inject constructor(
             AuthResult.Success(data = taskyNetwork.getAttendee(email))
         }
     }
+
+    override suspend fun leaveEvent(eventId: String): AuthResult<Unit> {
+        return authenticatedRetrofitCall(serializer) {
+            taskyNetwork.leaveEvent(eventId)
+            dao.leaveEvent(eventId)
```

Comment:
Here, `withContext(NonCancellable)` should also be needed. The API call can succeed and the coroutine could then be cancelled during `dao.leaveEvent`